[[outputs.nats]] panic: missing method Field

### Relevant telegraf.conf

```toml
[[outputs.nats]]
    alias = "output-nats-influxdb"

    servers = ["$NATS_SERVER_URL"]
    name = "catapult_nats_to_influxdb"
    subject = "catapult.influxdb_forward"
    data_format = "influx"
    flush_interval = "5s"
    tagexclude = ["processor", "direct_to_influx", "redis_expiration_*"]

    use_batch_format = false

    [outputs.nats.jetstream]
        name = "catapult"
        subjects = ["catapult.influxdb_forward"]
        retention = "workqueue"
```

### Logs from Telegraf

```text
panic: interface conversion: *metric.trackingMetric is not telegraf.TemplateMetric: missing method Field
```

### System info

Telegraf 1.36.1, Docker image docker.io/telegraf:1.36.1-alpine

### Docker

_No response_

### Steps to reproduce

I haven't been able to reproduce it on demand, but it happens consistently on our production environment. I'm hopeful you will have more knowledge of what's going on here to identify the issue.


### Expected behavior

Metrics should be published on NATs output

### Actual behavior

The telegraf process exits with exit_code 2.

### Additional info

We think the issue is related to this line: 

https://github.com/influxdata/telegraf/blob/3d03e5fb3170d783f45bfb01216408adcb4c94b7/plugins/outputs/nats/nats.go#L346

When enabling `use_batch_format`, that code block is skipped and the panic stops happening.

--- a/plugins/outputs/nats/nats.go
+++ b/plugins/outputs/nats/nats.go
@@ -341,9 +341,13 @@ func (n *NATS) Write(metrics []telegraf.Metric) error {
 		}
 	} else {
 		var subject bytes.Buffer
-		for i, m := range metrics {
+		for i, raw := range metrics {
+			m := raw
+			if wm, ok := m.(telegraf.UnwrappableMetric); ok {
+				m = wm.Unwrap()
+			}
 			subject.Reset()
-			if err := n.tplSubject.Execute(&subject, m.(telegraf.TemplateMetric)); err != nil {
+			if err := n.tplSubject.Execute(&subject, m); err != nil {
 				return fmt.Errorf("failed to execute subject template: %w", err)
 			}
 			sub := subject.String()

